src/home_robot/home_robot/utils/image.py

```diff
@@ -193,3 +200,53 @@ def rotate_image(imgs: List[np.ndarray]) -> List[np.ndarray]:
     """stretch specific routine to flip and rotate sideways images for normal viewing"""
     imgs = [np.rot90(np.fliplr(np.flipud(x))) for x in imgs]
     return imgs
+
+
+def build_mask(
+    target: Tensor, val: float = 0.0, tol: float = 1e-3, mask_extra_radius: int = 5
+) -> Tensor:
+    """Build mask where all channels are (val - tol) <= target <= (val + tol)
+        Optionally, dilate by mask_extra_radius
+
+    Args:
+        target (Tensor): [B, N_channels, H, W] input tensor
+        val (float, optional): Value to use for masking. Defaults to 0.0.
+        tol (float, optional): Tolerance for mask. Defaults to 1e-3.
+        mask_extra_radius (int, optional): Dilate by mask_extra_radius pix . Defaults to 5.
+
+    Returns:
+        _type_: Mask of shape target.shape
+    """
+    assert target.ndim == 4, target.shape
```

Comment:
better error message here, maybe f"target should be of shape [B, N_channels, H, W], was {target.shape}"